### PR TITLE
feat: allow custom image fit property

### PR DESCRIPTION
**Pull request by Xstoudi**
Meh:
image

Yeah:
image

Not really sure if it's the best way to achieve it.

### CODE REVIEW

File: gatsby-theme-portfolio-minimal/src/components/Project/index.tsx [hunk 1]
```diff
@@ -32,6 +32,8 @@ interface ProjectProps {
 export function Project(props: ProjectProps): React.ReactElement {
     const isDesktopBreakpoint = useMediaQuery('(min-width: 992px)');
 
+    console.log(props.data.image);
+
```
Comment (konstantinmuenster) [inline gatsby-theme-portfolio-minimal/src/components/Project/index.tsx:36]: This could be removed 😄

**Review by konstantinmuenster — CHANGES_REQUESTED** [review]
Only a small little change to make! Will need to test it locally, but looks good overall!

Thanks for contributing! Really appreciate it 👏☺️

File: gatsby-theme-portfolio-minimal/src/components/ArticleCard/index.tsx [hunk 2]
```diff
@@ -38,6 +38,7 @@ export function ArticleCard(props: ArticleCardProps): React.ReactElement {
                         <GatsbyImage
                             className={classes.ImageWrapper}
                             imgClassName={classes.Image}
+                            objectFit={props.data.image.objectFit || 'cover'}
```
Comment (konstantinmuenster) [inline gatsby-theme-portfolio-minimal/src/components/ArticleCard/index.tsx:41]: Can we remove the fallback `|| 'cover'`?

So you are able to specify no "object-fit" at all if you don't want to

File: gatsby-theme-portfolio-minimal/src/components/Project/index.tsx [hunk 3]
```diff
@@ -78,6 +80,7 @@ export function Project(props: ProjectProps): React.ReactElement {
                     <GatsbyImage
                         className={classes.ProjectImageWrapper}
                         imgClassName={classes.ProjectImage}
+                        objectFit={props.data.image.objectFit || 'cover'}
```
Comment (konstantinmuenster) [inline gatsby-theme-portfolio-minimal/src/components/Project/index.tsx:83]: Can we remove the fallback `|| 'cover'`?

So you are able to specify no "object-fit" at all if you don't want to

File: gatsby-theme-portfolio-minimal/src/components/Project/index.tsx [hunk 4]
```diff
@@ -87,6 +90,7 @@ export function Project(props: ProjectProps): React.ReactElement {
                 <GatsbyImage
                     className={classes.ProjectImageWrapper}
                     imgClassName={classes.ProjectImage}
+                    objectFit={props.data.image.objectFit || 'cover'}
```
Comment (konstantinmuenster) [inline gatsby-theme-portfolio-minimal/src/components/Project/index.tsx:93]: Can we remove the fallback `|| 'cover'`?

So you are able to specify no "object-fit" at all if you don't want to

**Review by konstantinmuenster — CHANGES_REQUESTED** [review]
Just tested it locally - works great! Could you have a look at the small comments I made, thanks!

And it would be great to add this new feature to the `gatsby-theme-portfolio-minimal/README.md` 🙏

**Comment by Xstoudi** [comment]
I'm not really sure about what to write in the README for that ahah, I'm open to suggestions.

**Comment by konstantinmuenster** [comment]
> I'm not really sure about what to write in the README for that ahah, I'm open to suggestions.

hm, true. There is currently no section that really fits. You could instead add the new prop to all section boilerplates files you find [in here](https://github.com/konstantinmuenster/gatsby-theme-portfolio-minimal/tree/main/gatsby-theme-portfolio-minimal/content/sections). Just want to make sure that new users know this prop exists.

So, for example, for the `contact.example.json`:

```json
{
    "name": "",
    "email": "",
    "description": "",
    "image": {
        "src": "",
        "alt": "",
        "objectFit": ""
    },
    "socialProfiles": {
        "from": [""],
        "showIcons": true
    }
}
```

**Comment by Xstoudi** [comment]
Make sense, I did it, let me know if anything else is needed.

**Comment by konstantinmuenster** [comment]
Thank you very much for contributing! :)

**Pull request merged**